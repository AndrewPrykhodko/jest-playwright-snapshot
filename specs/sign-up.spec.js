const open = require('../test/hooks').open
const signupPage = require('../book/signup.page')
const signUp = require('../facades/signup')
const { users } = require('../data/signup-user.json')
const valid = require('../data/generated/user.js.json')

// mapping of static invalid scenarios
users.map(user =>
    describe(`signup for ${user.fields.name}`, () => {
        beforeAll(async () => {
            await open(signupPage.url)
        })

        Object.entries(user.fields).map(([field, value]) => {
            signUp.fill(field, value)
        })

        Object.entries(user.checkbox).map(([checkbox, value]) => {
            signUp.checkbox(checkbox, value)
        })

        signUp.submit(user)
    })
)

// valid scenario with dynamically generated data
describe('signup for valid', () => {
    beforeAll(async () => {
        await open(signupPage.url)
    })

    Object.entries(valid.fields).map(([field, value]) => {
        signUp.fill(field, value)
    })

    Object.entries(valid.checkbox).map(([checkbox, value]) => {
        signUp.checkbox(checkbox, value)
    })

    signUp.submit(valid)
})
