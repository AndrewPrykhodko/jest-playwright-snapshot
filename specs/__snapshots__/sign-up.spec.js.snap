// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`signup for cliff submit 1`] = `
Object {
  "color": "rgb(255, 208, 47)",
  "font": "Formular, Helvetica, Arial, serif",
  "size": "14.9333px",
  "text": "Weak password",
}
`;

exports[`signup for cliff submit 2`] = `
Object {
  "color": "rgb(255, 101, 117)",
  "font": "Formular, Helvetica, Arial, serif",
  "size": "14.9333px",
  "text": "Please agree with the Terms to sign up.",
}
`;

exports[`signup for petra submit 1`] = `
Object {
  "color": "rgb(255, 101, 117)",
  "font": "Formular, Helvetica, Arial, serif",
  "size": "14.9333px",
  "text": "Enter your email address.",
}
`;

exports[`signup for petra submit 2`] = `
Object {
  "color": "rgb(255, 101, 117)",
  "font": "Formular, Helvetica, Arial, serif",
  "size": "14.9333px",
  "text": "Enter your password.
			",
}
`;

exports[`signup for rock submit 1`] = `
Object {
  "color": "rgb(255, 101, 117)",
  "font": "Formular, Helvetica, Arial, serif",
  "size": "14.9333px",
  "text": "Enter a valid email address.",
}
`;

exports[`signup for rock submit 2`] = `
Object {
  "color": "rgb(255, 101, 117)",
  "font": "Formular, Helvetica, Arial, serif",
  "size": "14.9333px",
  "text": "Please use 8+ characters for secure password.",
}
`;

exports[`signup for rock submit 3`] = `
Object {
  "color": "rgb(255, 101, 117)",
  "font": "Formular, Helvetica, Arial, serif",
  "size": "14.9333px",
  "text": "Please agree with the Terms to sign up.",
}
`;

exports[`signup for stone submit 1`] = `
Object {
  "color": "rgb(255, 101, 117)",
  "font": "Formular, Helvetica, Arial, serif",
  "size": "14.9333px",
  "text": "Sorry, this email is already registered",
}
`;

exports[`signup for valid submit 1`] = `
Object {
  "color": "rgb(5, 0, 56)",
  "font": "Spoof, Helvetica, Arial, serif",
  "size": "34.1333px",
  "text": "Check your email",
}
`;

exports[`signup for valid submit 2`] = `
Object {
  "color": "rgb(5, 0, 56)",
  "font": "Formular, Helvetica, Arial, serif",
  "size": "19.2px",
  "text": "",
}
`;
